src/core/qgsrelationmanager.cpp

```diff
@@ -25,9 +25,12 @@ QgsRelationManager::QgsRelationManager( QgsProject *project )
   : QObject( project )
   , mProject( project )
 {
-  connect( project, &QgsProject::readProject, this, &QgsRelationManager::readProject );
-  connect( project, &QgsProject::writeProject, this, &QgsRelationManager::writeProject );
-  connect( project, &QgsProject::layersRemoved, this, &QgsRelationManager::layersRemoved );
+  if ( mProject )
+  {
+    connect( project, &QgsProject::readProject, this, &QgsRelationManager::readProject );
```

Comment:
Isn't QgsRelationManager always instantiated by QgsProject (and then owned by and a child of this project)?
It seems weird if the project can be a `nullptr` in this scenario. Did I miss something here?

Reply:
normally yes, but [qmlplugindump](http://doc.qt.io/qtcreator/creator-qml-modules-with-plugins.html#generating-qmltypes-files) uses default constructor to initialize types to be able to extract some info from it.

Reply:
Out of curiosity, when is this executed, build time or run time?

Reply:
it is executed after target qgis_quick_plugin is build, see 
https://github.com/PeterPetrik/QGIS/blob/qgsquick_1_canvas/src/quickgui/plugin/CMakeLists.txt#L106

Reply:
And if it's registered as uncreatable type (as done originally in QField)?

Reply:
yep, lets register just uncreatable type.